refactor(linter): Use DefaultRuleConfig for more rules.

connorshea opened the pull request:
This allows us to simplify the code a decent amount in a lot of rules, and rely on the config schema defined for the config docs.

connorshea commented on crates/oxc_linter/src/rules/jsx_a11y/aria_role.rs:
```diff
@@ -176,13 +157,13 @@ fn test() {
     use crate::tester::Tester;
 
     fn ignore_non_dom_schema() -> serde_json::Value {
-        serde_json::json!([2,{
```
These are necessary to remove, otherwise the rule config doesn't work. I don't entirely know why this was working previously?

camc314 commented:
### Merge activity

* **Nov 29, 3:24 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 29, 3:24 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

connorshea committed:
refactor(linter): Use DefaultRuleConfig for more rules. (#16267)

This allows us to simplify the code a decent amount in a lot of rules, and rely on the config schema defined for the config docs.
